tests/unit/amo/components/TestInstallWarning.js

```diff
@@ -141,11 +151,43 @@ describe(__filename, () => {
 
       expect(component.instance().couldShowWarning()).toEqual(false);
     });
+
+    it('returns false if the WrongPlatformWarning would be shown', () => {
+      const _correctedLocationForPlatform = sinon.stub().returns('/some/path/');
+
+      const component = renderWithWarning({ _correctedLocationForPlatform });
+
+      expect(component.instance().couldShowWarning()).toEqual(false);
+    });
+
+    it('calls _correctedLocationForPlatform with clientApp, location and userAgentInfo', () => {
+      const clientApp = CLIENT_APP_ANDROID;
```

Comment:
This relies on the default value. You should pass into `dispatchClientMetadata` explicitly instead.